JOSESwift/Sources/CryptoImplementation/AES.swift

```diff
@@ -97,37 +123,122 @@ internal struct AES {
     ///   - initializationVector: The initial block.
     /// - Returns: The plain text (decrypted cipher text).
     /// - Throws: `AESError` if any error occurs during decryption.
-    static func decrypt(cipherText: Data, with decryptionKey: Data, using algorithm: ContentEncryptionAlgorithm, and initializationVector: Data) throws -> Data {
+    static func decrypt(
+        cipherText: Data,
+        with decryptionKey: Data,
+        using algorithm: ContentEncryptionAlgorithm,
+        and initializationVector: Data
+    ) throws -> Data {
         switch algorithm {
         case .A256CBCHS512, .A128CBCHS256:
             guard algorithm.checkAESKeyLength(for: decryptionKey) else {
                 throw AESError.keyLengthNotSatisfied
             }
 
-            let decrypted = try aes(
+            let decrypted = ccAESCBCCrypt(
                 operation: CCOperation(kCCDecrypt),
                 data: cipherText, key: decryptionKey,
                 algorithm: algorithm.ccAlgorithm,
                 initializationVector: initializationVector,
                 padding: CCOptions(kCCOptionPKCS7Padding)
             )
 
-            guard decrypted.status == UInt32(kCCSuccess) else {
-                throw AESError.decryptingFailed(description: "Decryption failed with CryptoStatus: \(decrypted.status).")
+            guard
+                let plaintext = decrypted.data,
+                decrypted.status == UInt32(kCCSuccess)
+            else {
+                throw AESError.decryptingFailed(description: "Decryption failed with status: \(decrypted.status).")
+            }
+
+            return plaintext
+        }
+    }
+
+    /// Encrypts the given raw key using AES key wrap.
+    /// - Parameters:
+    ///   - rawKey: The raw key to encrypt
+    ///   - keyEncryptionKey: The key used to encypt the raw key
+    ///   - algorithm: The algorithm to use for AES key wrap
+    static func keyWrap(rawKey: Data, keyEncryptionKey: Data, algorithm: KeyManagementAlgorithm) throws -> Data {
+        switch algorithm {
+        case .A128KW, .A192KW, .A256KW:
```

Comment:
What do you think of refactoring this to a guard? It's a little clunkier, but you can drop an indentation level, and have the "else" case much closer to it's check.
```swift
let okAlgorithms = [.A128KW, .A192KW, .A256KW]
guard okAlgorithms.contains(algorithm) else {
     throw AESError.invalidAlgorithm 
}
```

Reply:
Good point. There might be other key wrap algorithms in the future where a switch will make sense but we don't need to care about them now. Refactored to guard. ✅